pkgs/tools/misc/nix-direnv/default.nix

```diff
@@ -26,6 +26,9 @@ stdenv.mkDerivation rec {
   installPhase = ''
     runHook preInstall
     install -m500 -D direnvrc $out/share/nix-direnv/direnvrc
+
+    # Allows NixOS to set DIRENV_CONFIG to /run/current-system/sw/share/direnv and have direnv load this
```

Comment:
In that case, the module will also have to handle user settings since DIRENV_CONFIG is also used to locate the `direnv.toml`.

Alternatively, it could make sense to change direnv so it also loads all the libraries in 
`share/direnv/lib/*.sh`, following the XDG spec?

Reply:
I think that, given that direnv already reads `${XDG_CONFIG_HOME:-$HOME/.config}/direnv/lib/*.sh` it makes perfect sense for it to read the equivalent system directory, does it not?

Reply:
Yeah I think so too, I quote the [XDG Base Directory Specification](https://specifications.freedesktop.org/basedir-spec/basedir-spec-latest.html):

> `$XDG_CONFIG_DIRS` defines the preference-ordered set of base directories to search for configuration files in addition to the `$XDG_CONFIG_HOME` base directory. The directories in `$XDG_CONFIG_DIRS` should be seperated with a colon ':'.
>
> If `$XDG_CONFIG_DIRS` is either not set or empty, a value equal to /etc/xdg should be used.

Reply:
I created a direnv PR for this: https://github.com/direnv/direnv/pull/990